fvm/handler/contract.go

```diff
@@ -55,8 +59,15 @@ func (h *ContractHandler) GetContract(address runtime.Address, name string) (cod
 func (h *ContractHandler) SetContract(address runtime.Address, name string, code []byte, signingAccounts []runtime.Address) (err error) {
 	// check if authorized
 	if !h.isAuthorized(signingAccounts) {
-		err = errors.NewOperationAuthorizationErrorf("SetContract", "setting contracts requires authorization from specific accounts")
-		return fmt.Errorf("setting contract failed: %w", err)
+		// check if there's an audit voucher for the contract
+		voucherUsed, err := h.useContractAuditVoucher(address, code)
+		if err != nil {
+			return fmt.Errorf("setting contract failed: (useContractAuditVoucher) %w", err)
```

Comment:
@janezpodhostnik did we check that errors that are passed to this method are not considered fatal?

Reply:
Fixed here https://github.com/onflow/flow-go/pull/1826/commits/47b3651096243fe7ddd6cf76fa9688e19734576e